presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchUtils.java

```diff
@@ -0,0 +1,126 @@
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.StandardTypes.ARRAY;
+import static com.facebook.presto.spi.type.StandardTypes.MAP;
+import static com.facebook.presto.spi.type.StandardTypes.ROW;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static io.airlift.slice.Slices.utf8Slice;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public final class ElasticsearchUtils
+{
+    private ElasticsearchUtils() {}
+
+    public static Block serializeObject(Type type, BlockBuilder builder, Object object)
+    {
+        if (ROW.equals(type.getTypeSignature().getBase())) {
+            return serializeStruct(type, builder, object);
+        }
+        else if (MAP.equals(type.getTypeSignature().getBase()) || ARRAY.equals(type.getTypeSignature().getBase())) {
+            throw new RuntimeException("Type not supported in Elasticsearch: " + type.getDisplayName());
+        }
+        else {
+            serializePrimitive(type, builder, object);
+            return null;
+        }
+    }
+
+    private static Block serializeStruct(Type type, BlockBuilder builder, Object object)
+    {
+        if (object == null) {
+            requireNonNull(builder, "parent builder is null").appendNull();
+            return null;
+        }
+
+        List<Type> typeParameters = type.getTypeParameters();
+        BlockBuilder currentBuilder;
+
+        boolean builderSynthesized = false;
+        if (builder == null) {
+            builderSynthesized = true;
+            builder = type.createBlockBuilder(null, 1);
+        }
+        currentBuilder = builder.beginBlockEntry();
+
+        for (int i = 0; i < typeParameters.size(); i++) {
+            Optional<String> fieldName = type.getTypeSignature().getParameters().get(i).getNamedTypeSignature().getName();
+            String name = "";
+            if (fieldName.isPresent()) {
+                name = fieldName.get().toLowerCase(ENGLISH);
+            }
+            Object value = ((Map) object).get(name);
+            serializeObject(typeParameters.get(i), currentBuilder, value);
+        }
+
+        builder.closeEntry();
+        if (builderSynthesized) {
+            return (Block) type.getObject(builder, 0);
+        }
+        else {
+            return null;
+        }
+    }
+
+    private static void serializePrimitive(Type type, BlockBuilder builder, Object object)
+    {
+        requireNonNull(builder, "parent builder is null");
+
+        if (object == null) {
+            builder.appendNull();
+            return;
+        }
+
+        if (type.equals(BOOLEAN)) {
+            type.writeBoolean(builder, (Boolean) object);
+            return;
```

Comment:
since you have a chain of if/else's you don't need to return in each condition.

Reply:
yep, will fix